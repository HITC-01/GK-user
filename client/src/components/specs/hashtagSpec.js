import React from 'react';
import renderer from 'react-test-renderer';
import HashTag from '../hashtag.jsx';
import { shallow, render } from '../../enzyme';


describe('<HashTag />', () => {
  test('Should render correctly', () => {
    const component = renderer.create(<HashTag />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('render correct component prop type', () => {
    const component = shallow(<HashTag hash="testing" />);

    expect(component.find('.up-hash').exists()).toBe(true);
  });

  test('Should render matching element', () => {
    const component = shallow(<HashTag hash="hashing" />);

    expect(component.containsMatchingElement(<div><span className="up-hash">#hashing</span></div>)).toBeTruthy();
  });
});
