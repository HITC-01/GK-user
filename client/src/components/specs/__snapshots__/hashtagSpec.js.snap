// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HashTag /> Should render correctly 1`] = `
<div>
  <span
    className="up-hash"
  >
    #
    Songs
  </span>
</div>
`;
